strawberry/relay/fields.py

```diff
@@ -0,0 +1,553 @@
+    # This init parameter is used by pyright to determine whether this field
+    # is added in the constructor or not. It is not used to change
+    # any behavior at the moment.
+    init: Literal[True, False, None] = None,
+) -> Any:
+    """Annotate a property to create a relay query field.
+
+    Examples:
+        Annotating something like this:
+
+        >>> @strawberry.type
+        >>> class X:
+        ...     some_node: SomeType = relay.node(description="ABC")
+
+        Will produce a query like this that returns `SomeType` given its id.
+
+        ```
+        query {
+            someNode (id: ID) {
+                id
+                ...
+            }
+        }
+        ```
+
+    """
+    return NodeField(
+        python_name=None,
+        graphql_name=name,
+        type_annotation=None,
+        description=description,
+        is_subscription=is_subscription,
+        permission_classes=permission_classes or [],
+        deprecation_reason=deprecation_reason,
+        default=default,
+        default_factory=default_factory,
+        metadata=metadata,
+        directives=directives or (),
+    )
+
+
+@overload
+def connection(
+    *,
+    resolver: Callable[[], _T],
+    name: Optional[str] = None,
+    is_subscription: bool = False,
+    description: Optional[str] = None,
+    init: Literal[False] = False,
+    permission_classes: Optional[List[Type[BasePermission]]] = None,
+    deprecation_reason: Optional[str] = None,
+    default: Any = dataclasses.MISSING,
+    default_factory: Union[Callable[..., object], object] = dataclasses.MISSING,
+    metadata: Optional[Mapping[Any, Any]] = None,
+    directives: Optional[Sequence[object]] = (),
+    graphql_type: Optional[Any] = None,
+) -> _T:
+    ...
+
+
+@overload
+def connection(
+    *,
+    name: Optional[str] = None,
+    is_subscription: bool = False,
+    description: Optional[str] = None,
+    init: Literal[True] = True,
+    permission_classes: Optional[List[Type[BasePermission]]] = None,
+    deprecation_reason: Optional[str] = None,
+    default: Any = dataclasses.MISSING,
+    default_factory: Union[Callable[..., object], object] = dataclasses.MISSING,
+    metadata: Optional[Mapping[Any, Any]] = None,
+    directives: Optional[Sequence[object]] = (),
+    graphql_type: Optional[Any] = None,
+) -> Any:
+    ...
+
+
+@overload
+def connection(
+    resolver: Union[StrawberryResolver, Callable, staticmethod, classmethod],
+    *,
+    name: Optional[str] = None,
+    is_subscription: bool = False,
+    description: Optional[str] = None,
+    permission_classes: Optional[List[Type[BasePermission]]] = None,
+    deprecation_reason: Optional[str] = None,
+    default: Any = dataclasses.MISSING,
+    default_factory: Union[Callable[..., object], object] = dataclasses.MISSING,
+    metadata: Optional[Mapping[Any, Any]] = None,
+    directives: Optional[Sequence[object]] = (),
+    graphql_type: Optional[Any] = None,
+) -> ConnectionField:
```

Comment:
this might be annoying, and I'm happy to do it for, but would be possible to add some pyright tests?

I think there's some small issues with typing when using strict mode:

<img width="769" alt="CleanShot 2023-02-09 at 23 23 35@2x" src="https://user-images.githubusercontent.com/667029/217961441-5331b26d-a841-4498-92fb-babdecd513cc.png">

Reply:
Sure, I can totally do that! :)

Does that only happen in strict mode? I use pyright in basic mode (with some extra checks enabled) and did not see that before. Will write some tests and fix any typing issues that strict mode shows :)

Reply:
yup!